src/App.tsx

```diff
@@ -1,18 +1,51 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const total = 42;
+const items = getNumbers(1, total)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const indexOfLastItem = useMemo(() => {
+    const index = currentPage * perPage;
+
+    if (index > total) {
+      return total;
+    }
+
+    return index;
```

Comment:
You can use a ternary operator here 
```suggestion
return index > total ? total : index;
```